components/elements/frontPage/LoginForm.js

```diff
@@ -1,121 +1,115 @@
-    (async () => {
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [state, dispatch] = useReducer(authReducer, initialState);
+
+  const onSubmit = useCallback(
+    (evt) => {
+      evt.preventDefault();
+      (async () => {
         try {
-          dispatch({type: 'INIT'});
-          const response = await slsFetch('/api/login', {
-            method: 'POST',
+          dispatch({ type: INIT });
+          const response = await slsFetch("/api/login", {
+            method: "POST",
             headers: {
-              'Content-type': 'application/json'
+              "Content-type": "application/json",
             },
-            body: JSON.stringify({email, password})
+            body: JSON.stringify({ email, password }),
           });
           const { message, homepage } = await response.json();
-          dispatch({type: 'SUCCESS'});
+          dispatch({ type: SUCCESS });
           router.push(`/admin`);
-          
-        } catch(ex) {
-          console.error(ex);  
-          dispatch({type: 'ERROR'});
+        } catch (ex) {
+          console.error(ex);
+          dispatch({ type: ERROR });
         } finally {
-          dispatch({type: 'CLEANUP'});
+          dispatch({ type: CLEANUP });
         }
-      //}
-    })();
-  }, [email, password, router]);
-  
+        //}
+      })();
+    },
+    [email, password, router]
+  );
+
   const errorBox = useRef();
   const successBox = useRef();
-  
- useFadeEffect(errorBox, [state.submitted, state.isError]);
- useFadeEffect(successBox, [state.submitted, state.isLoginSuccessful]);
-  
+
+  useFadeEffect(errorBox, [state.submitted, state.isError]);
+  useFadeEffect(successBox, [state.submitted, state.isLoginSuccessful]);
+
   return (
-    <div className='container_login_form'>
-				<div className='img_login_logo img-responsive'>	
-          <Image placeholder='blur'
-                blurDataURL="/logo-180x180.png"
-                src='/logo-180x180.png'
-                alt='cms-logo'
-                width={200} height={90}
-          />
-  			</div>
-			  <h4> Welcome to KlaudSol CMS!  </h4>
-				
-				<div className='form_login'>
-        <div ref={errorBox} className="alert alert-danger useFadeEffect px-3 pt-3 pb-2 mb-0 mt-3"> <p>Incorrect username and/or password.</p> </div>
-        <div ref={successBox} className="alert alert-success useFadeEffect px-3 pt-3 pb-2 mb-0 mt-3"> <p>Welcome back! Logging you in...</p> </div>
-              		<label className='mb-2 mt-3'>Email</label>
-              		<input type='email' className='input_login' autoComplete="email" onChange={e => setEmail(e.target.value)}  />
+    <div className="container_login_form">
+      <div className="img_login_logo img-responsive">
+       {logo?.link && <Image
+          placeholder="blur"
+          blurDataURL={logo?.link}
+          src={logo?.link}
+          alt="cms-logo"
+          width={200}   
+          height={90}
+        />}
```

Comment:
If the logo is not provided, it should fall back to the original Klaudsol logo. It shouldn't be blank.
```suggestion
       {logo?.link && <Image
          placeholder="blur"
          blurDataURL={logo?.link}
          src={logo?.link}
          alt="cms-logo"
          width={200}   
          height={90}
        />}
        
        {!logo.link && (
          <Image placeholder='blur'
                blurDataURL="/logo-180x180.png"
                src='/logo-180x180.png'
                alt='cms-logo'
                width={200} height={90}
          />        
        )}
```